cpp/core/memory/MemoryAllocator.cc

```diff
@@ -131,11 +131,13 @@ bool StdMemoryAllocator::ReallocateAligned(void* p, uint16_t alignment, int64_t
   if (new_size <= 0) {
     return false;
   }
-  void* reallocated_p = std::realloc(p, new_size);
+  void* reallocated_p = std::malloc(new_size);
```

Comment:
Why change realloc to malloc and free?

Reply:
This was a bug. `std::realloc` doesn't require manual copy and the `out` pointer was not updated which was wrong in previous code